src/OpenSage.FileFormats.Big/BigArchive.cs

```diff
@@ -118,6 +134,136 @@ public BigArchiveEntry GetEntry(string entryName)
             _entriesDictionary.TryGetValue(entryName, out var result);
             return result;
         }
+
+        internal void DeleteEntry(BigArchiveEntry entry)
+        {
+            _entries.Remove(entry);
+            WriteToDisk(true);
+        }
+
+        private long CalculateContentSize()
+        {
+            long contentSize = 0;
+            foreach (var entry in _entries)
+            {
+                contentSize += entry.Length;
+            }
+
+            return contentSize;
+        }
+
+        private int CalculateTableSize()
+        {
+            int tableSize = 0;
+            foreach (var entry in _entries)
+            {
+                // Each entry has 4 bytes for the offset + 4 for size
+                tableSize += 8;
```

Comment:
maybe use a const ENTRY_SIZE here, that would also make the comment above obsolete